### PR TITLE
Implemented Grade feature

### PR DESCRIPTION
**Pull request by tristyxxnana**
This PR consists of:
1. Grade feature implementation
- grade add
- grade list
- grade graph
2. Removed outdated test due to modification of codes

Will include tests in the future PR.

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/GradeAddCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package seedu.address.logic.parser;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import seedu.address.logic.commands.GradeAddCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.GradeAddCommandParser.*;
```
Comment (harriuscai) [inline src/test/java/seedu/address/logic/parser/GradeAddCommandParserTest.java]: Avoid using wildcard `*` imports

File: src/test/java/seedu/address/logic/parser/GradebookAddCommandParserTest.java [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 import static org.junit.Assert.assertNotNull;
 import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.*;
```
Comment (harriuscai) [inline src/test/java/seedu/address/logic/parser/GradebookAddCommandParserTest.java]: Avoid wildcard `*` imports

File: src/main/java/seedu/address/model/grades/GradesManager.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        if (studentMarks >= 0 && studentMarks <= 100) {
+            isMarksValid = true;
+        }
+        return isMarksValid;
+    }
+
+    /**
+     This method checks if student is enrolled to module.
+     */
+    public boolean isStudentEnrolledToModule (String moduleCode, String adminNo) {
```
Comment (harriuscai) [inline src/main/java/seedu/address/model/grades/GradesManager.java]: Enrolling of students into a module is part of my feature, so this API should be provided by my manager class. Please switch to it when I have pushed it out

File: src/main/java/seedu/address/logic/commands/GradebookEditCommand.java [hunk 4]
```diff
@@ -51,7 +54,9 @@ public CommandResult execute (Model model, CommandHistory history) throws Comman
         if (gradebook == null) {
             return new CommandResult(MESSAGE_FIND_FAIL);
         }
-
+        if (newGradeComponentName.equals("") && newMaxMarks == 0 && newWeightage == 0) {
```
[the remark this answers is not in the document]
Comment (tristyxxnana) [reply src/main/java/seedu/address/logic/commands/GradebookEditCommand.java]: will do so in the next version release, possibly v1.4

File: src/main/java/seedu/address/model/grades/GradesManager.java (hunk 3 above)
Comment (tristyxxnana) [reply src/main/java/seedu/address/model/grades/GradesManager.java]: Will try to make use of this by v1.3, if not will be pushed to v1.4 release. Thanks

**Pull request merged**